### PR TITLE
t-SNE patching

### PR DESCRIPTION
**Pull request by agorshk**
Related [PR to oneDAL](https://github.com/oneapi-src/oneDAL/pull/1822)

### CODE REVIEW

**Comment by agorshk** [comment]
Look's good

**Comment by Pahandrovich** [comment]
last CI 
http://intel-ci.intel.com/ec3b3490-79eb-f1a0-abb2-8cdcd4b723a1

**Review by napetrov — APPROVED** [review]

**Comment by Pahandrovich** [comment]
@agorshk, @vlad-nazarov, @KalyanovD, @itearsl, please, review this pull request

File: src/daal4py.cpp [hunk 1]
```diff
@@ -871,12 +868,7 @@ double c_roc_auc_score(data_or_file & y_true, data_or_file & y_test)
     auto table_true = get_table(y_true);
     auto table_test = get_table(y_test);
     auto type       = (*table_test->getDictionary())[0].indexType;
-    if (type == daal::data_management::data_feature_utils::DAAL_FLOAT64 ||
-        type == daal::data_management::data_feature_utils::DAAL_INT64_S ||
-        type == daal::data_management::data_feature_utils::DAAL_INT64_U ||
-        type == daal::data_management::data_feature_utils::DAAL_FLOAT32 ||
-        type == daal::data_management::data_feature_utils::DAAL_INT32_S ||
-        type == daal::data_management::data_feature_utils::DAAL_INT32_U)
+    if (type == daal::data_management::data_feature_utils::DAAL_FLOAT64 || type == daal::data_management::data_feature_utils::DAAL_INT64_S || type == daal::data_management::data_feature_utils::DAAL_INT64_U || type == daal::data_management::data_feature_utils::DAAL_FLOAT32 || type == daal::data_management::data_feature_utils::DAAL_INT32_S || type == daal::data_management::data_feature_utils::DAAL_INT32_U)
```
Comment (vlad-nazarov) [inline src/daal4py.cpp:871]: Is it right autoformatting?

File: src/daal4py.cpp [hunk 2]
```diff
@@ -897,3 +889,32 @@ void c_generate_shuffled_indices(data_or_file & idx, data_or_file & random_state
 #else
 #endif
 }
+
+void c_tsne_gradient_descent(data_or_file & init, data_or_file & p, data_or_file & size_iter, data_or_file & params, data_or_file & results, char dtype)
+{
+#if __INTEL_DAAL__ == 2021 && INTEL_DAAL_VERSION >= 20210400
```
Comment (vlad-nazarov) [inline src/daal4py.cpp:895]: Need to update INTEL_DAAL_VERSION

File: src/daal4py.h [hunk 3]
```diff
@@ -55,6 +55,9 @@ using daal::services::LibraryVersionInfo;
 #if __INTEL_DAAL__ == 2021 && INTEL_DAAL_VERSION >= 20210200
     #include "data_management/data/internal/roc_auc_score.h"
 #endif
+#if __INTEL_DAAL__ == 2021 && INTEL_DAAL_VERSION >= 20210500
```
Comment (vlad-nazarov) [inline src/daal4py.h:58]: Need to update INTEL_DAAL_VERSION

**Review by vlad-nazarov — APPROVED** [review]

File: src/daal4py.cpp (hunk 1 above)
Comment (Pahandrovich) [reply src/daal4py.cpp:871]: clang-format fixed it, I'll get it back. Thanks

File: daal4py/sklearn/manifold/__init__.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 #===============================================================================
-# Copyright 2014 Intel Corporation
+# Copyright 2020 Intel Corporation
```
Comment (agorshk) [inline daal4py/sklearn/manifold/__init__.py:3]: 2020-2022 ?

**Review by agorshk — COMMENTED** [review]
Looks good

File: daal4py/sklearn/manifold/__init__.py (hunk 4 above)
Comment (Pahandrovich) [reply daal4py/sklearn/manifold/__init__.py:3]: Now we have to specify only the year of adding the file

Comment (napetrov) [reply daal4py/sklearn/manifold/__init__.py:3]: we can use oldest date only - there is no need for interval. 

So date should reflect year of file/content creation

**Comment by Pahandrovich** [comment]
http://intel.intel.com/ec8f20b2-8923-f182-af7c-8cdcd4b723a1

**Comment by Pahandrovich** [comment]
http://intel-ci.intel.com/ec8f65a3-9e93-f14e-959a-8cdcd4b723a1

**Pull request merged**